zulipterminal/model.py

```diff
@@ -727,8 +727,10 @@ def _handle_message_event(self, event: Event) -> None:
             self.controller.update_screen()
             self._notified_user_of_notification_failure = True
 
+        self.index = index_messages([message], self, self.index)
+        if 'read' not in message['flags']:
+            set_count([message['id']], self.controller, 1)
```

Comment:
minor: follow this with a newline, and possibly add a comment to this block re updating counts and how the message needs to be in the index to be identified by `set_count`.

From other conversations, I wonder if we might update `set_count` to just accept the message(s) themselves to avoid an index dependence, but that's another discussion. That would avoid the coupling between these two actions (add to index, set count), as the message need not necessarily be in the index.